### ISSUE TITLE
PIF build errors when using language mode v5

### ISSUE THREAD

**Issue by glbrntt** [swiftlang/swift-package-manager#7958]
### Is it reproducible with SwiftPM command-line tools: `swift build`, `swift test`, `swift package` etc?

- [X] Confirmed reproduction steps with SwiftPM CLI. The description text _must_ include reproduction steps with either of command-line SwiftPM commands, `swift build`, `swift test`, `swift package` etc.

### Description

In grpc-swift we produce a fat binary of one of our products using `swift build --arch x86_64 --arch arm64`. The product uses the `.v5` Swift language mode but this fails to compile:

```
$ swift build --arch x86_64 --arch arm64
error: 'somepackage': Some of the Swift language versions used in target 'SomePackage' settings are supported. (given: [5], supported: [4.0, 4.2, 5.0, 6.0])
```

SwiftPM appears to treat `.v5` as "5" internally but is expecting "5.0" for PIF builds.

Changing `.swiftLanguageMode(.v5)` to `.swiftLanguageMode(.version("5.0"))` works for PIF builds but fails for a regular `swift build`.

Relatedly, there's a bug in the error message too: 

> Some of the Swift language versions used in target 'SomePackage' settings are supported.

should read 

> Some of the Swift language versions used in target 'SomePackage' settings are **un**supported.

### Expected behavior

Build should succeed.

### Actual behavior

Build fails with an error.

### Steps to reproduce

1. Download attached repro project
2. Run `swift build --arch x86_64 --arch arm64`

[SomePackage.zip](https://github.com/user-attachments/files/16960948/SomePackage.zip)

### Swift & OS version (output of `swift --version ; uname -a`)

Xcode 16 rc 1

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0

**Comment by CrazyFanFan** [comment]
Same issue
Xcode 16.0

**Comment by stackotter** [comment]
Same issue with Xcode 16.3

**Comment by plemarquand** [comment]
I can reproduce this with Xcode 16.3, but it looks to be fixed in the nightly toolchains.

**Comment by stackotter** [comment]
I'm getting the same behaviour with Xcode 16.4

```
Xcode 16.4
Build version 16F6

swift-driver version: 1.120.5 Apple Swift version 6.1.2 (swiftlang-6.1.2.1.2 clang-1700.0.13.5)
Target: arm64-apple-macosx15.0
```

**Comment by weissi** [comment]
`.v5` should work but `.version("5.0")` is incorrect, `5.0` isn't a language mode. I think this works as a workaround:

```
swiftSettings: [.swiftLanguageMode(.version("5"))]
```

**Comment by glbrntt** [comment]
> `.v5` should work but `.version("5.0")` is incorrect, `5.0` isn't a language mode. 

Correct, but SwiftPM mishandled this in previous Swift versions (6.0 and 6.1, I think).

> ... I think this works as a workaround:
> 
> ```
> swiftSettings: [.swiftLanguageMode(.version("5"))]
> ```

Unfortunately not for Xcode 16.4.
